morgoth/base/static/js/utils/apiFetch.js

```diff
@@ -0,0 +1,70 @@
+const API_ROOT = '/api/v1/';
+
+export default function apiFetch(url, options = {}) {
+  let dispatch;
+  let handleSuccess;
+  let handleError;
+
+  const headers = new Headers();
+  headers.append('Content-Type', 'application/json');
+
+  const settings = {
+    headers,
+    ...options,
+  };
+
+  // Convert `data` to `body` if necessary.
+  if ('data' in settings) {
+    if ('body' in settings) {
+      throw new Error('Only pass one of `settings.data` and `settings.body`.');
+    }
+
+    const method = (settings.method || 'get').toLowerCase();
+
+    if (method !== 'get' && method !== 'head') {
+      settings.body = JSON.stringify(settings.data);
+    }
+
+    delete settings.data;
```

Comment:
Probably isn't a good idea to just throw away `settings.data` silently if method is "get" or "head".